### PR TITLE
Add clarification document for the Twins sample

### PR DESCRIPTION
**Pull request by andyk-ms**
## Purpose

* Addressing issue reported: https://msazure.visualstudio.com/One/_workitems/edit/12845683
* Add clarification document for the Twins Sample
* Provide document links to standard twins interacts with IoT Hub and documentation for the details
* Provide a note for documents and links to specific conventions of JSON document

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
open twins sample readme
```

## What to Check
Verify that the following are valid
* Readme file shows on the sample page

### CODE REVIEW

**Review by brycewang-microsoft — APPROVED** [review]

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Device Twins Sample
+
+[![Documentation](../../../../docs/media/docs-link-buttons/azure-documentation.svg)](https://docs.microsoft.com/azure/iot-develop/)
+
+
+This sample code demonstrates device twins support interacts with IoT Hub.
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:6]:
```suggestion
This sample code demonstrates device twins support with IoT hub.
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+For more info about Device Twins:
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:8]:
```suggestion
For more info about device twins:  
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Device Twins Sample
+
+[![Documentation](../../../../docs/media/docs-link-buttons/azure-documentation.svg)](https://docs.microsoft.com/azure/iot-develop/)
+
+
+This sample code demonstrates device twins support interacts with IoT Hub.
+
+For more info about Device Twins:  
+- [Device Twin getting started](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-twin-getstarted)
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:9]:
```suggestion
- [Device twin getting started](https://docs.microsoft.com/azure/iot-hub/iot-hub-csharp-csharp-twin-getstarted)
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+- [Device Twins Developer Guide](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins)
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:10]:
```suggestion
- [Device Twins Developer Guide](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-device-twins)
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+> Note for PnP Device Twin Support with IoT Hub and IoT Central, there specific conventions of the JSON document to follow:
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:12]:
```suggestion
> Note for Plug and Play device twin support with IoT hub and IoT Central, there specific conventions of the JSON format to follow:
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+> - [PnP Device developer Guide](https://docs.microsoft.com/en-us/azure/iot-develop/concepts-developer-guide-device?pivots=programming-language-csharp)
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:13]:
```suggestion
> - [PnP device developer guide](https://docs.microsoft.com/azure/iot-develop/concepts-developer-guide-device?pivots=programming-language-csharp)
```

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+> - [PnP Device Samples](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/PnpDeviceSamples)
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:14]:
```suggestion
> - [PnP device samples](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/PnpDeviceSamples)
```

**Review by drwill-ms — COMMENTED** [review]
Just needs consistent casing, casing of IoT hub consistent with guidelines from the docs team, and removal of en-us from URLs.

File: iot-hub/Samples/device/TwinSample/readme.md (hunk 1 above)
Comment (andyk-ms) [reply iot-hub/Samples/device/TwinSample/readme.md:6]: Updated

**Pull request merged**